### PR TITLE
Add Scrolling

### PR DESCRIPTION
**Pull request by tavitenari**
Adding scrolling via a Camera class that locks on the player.

### CODE REVIEW

**Review by drmonkeysee — APPROVED** [review]

File: weapons/projectile.py [hunk 1]
```diff
@@ -27,9 +26,5 @@ def update(self, model, lag_scalar):
         self.x += (self.move_x + self.momentum_x) * lag_scalar
         self.y += (self.move_y + self.momentum_y) * lag_scalar
         radius = self.collider.radius
-        if (self.impacted
-            or self.x < -radius
-            or self.x > (state.SCREEN_WIDTH + radius)
-            or self.y < -radius
-            or self.y > (state.SCREEN_HEIGHT + radius)):
+        if self.impacted:  # TODO: If level size is finite, remove projectile if it travels outside level.
```
Comment (mtschaffer) [inline weapons/projectile.py:29]: Currently we have a finite height but infinite width.  You might want to keep the checks for removing the object if it goes outside the height bounds.

File: scene/level_one.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ def __init__(self):
 
         self.watermelon = Watermelon(seed_inventory=30)
         self.add_fg_element(self.watermelon)
+        camera.set_target(self.watermelon)
```
Comment (mtschaffer) [inline scene/level_one.py:36]: It's cool we can set the camera target.  Might make for some interesting camera tricks like following a seed or fight a boss from their perspective.

File: weapons/projectile.py (hunk 1 above)
Comment (mtschaffer) [reply weapons/projectile.py:29]: Oh, wait, I guess we also have an infinite y-axis as well.  Someone needs to make the background scroll in the Y direction too! :)

**Review by mtschaffer — APPROVED** [review]

File: weapons/projectile.py (hunk 1 above)
Comment (tavitenari) [reply weapons/projectile.py:29]: yeah, not sure! I didnt want to do too much and add a level size. We could just scroll on the x -axis with the camera. I think the vertical scrolling would nice too.

File: utils/sprite.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def __init__(self, sprite, effect=None, reaction=None):
 
     @property
     def center(self):
-        return int(self.sprite().x), int(self.sprite().y)
+        return self.sprite().x, self.sprite().y
```
Comment (drmonkeysee) [inline utils/sprite.py:37]: @mtschaffer you should make sure this doesn't introduce errors in collision detection

**Pull request merged**